### PR TITLE
Promote SubjectAccessReview to v1

### LINKED ISSUE #32709: SubjectAccessReview typos groups field name
The `Groups` field of a SubjectAccessReviewSpec is being serialized as `group`:
https://github.com/kubernetes/kubernetes/blob/b841a8bad3855c5c4b95c7dd2596e22018f9af68/pkg/apis/authorization/v1beta1/types.go#L119:

```
// Groups is the groups you're testing for.
Groups []string `json:"group,omitempty" protobuf:"bytes,4,rep,name=group"`
```

This is already documented as `group` for v1beta1, so we can't easily change it now, but should be fixed in the v1 version

### PR DESCRIPTION
**Pull request by liggitt**
We have multiple features that depend on this API:

SubjectAccessReview
- [webhook authorization](https://kubernetes.io/docs/admin/authorization/#webhook-mode)
- [kubelet delegated authorization](https://kubernetes.io/docs/admin/kubelet-authentication-authorization/#kubelet-authorization)
- add-on API server delegated authorization

The API has been in use since 1.3 in beta status (v1beta1) with negligible changes:
- Added a status field for reporting errors evaluating access
- A typo was discovered in the SubjectAccessReviewSpec Groups field name

This PR promotes the existing v1beta1 API to v1, with the only change being the typo fix to the groups field. (fixes https://github.com/kubernetes/kubernetes/issues/32709)

Because the API does not persist data (it is a query/response-style API), there are no data migration concerns.

This positions us to promote the features that depend on this API to stable in 1.7

cc @kubernetes/sig-auth-api-reviews @kubernetes/sig-auth-misc

```release-note
The authorization.k8s.io API group was promoted to v1
```

### CODE REVIEW

File: cmd/libs/go2idl/informer-gen/generators/packages.go [hunk 1]
```diff
@@ -63,14 +62,7 @@ func DefaultNameSystem() string {
...
 func generatedBy() string {
-	var cmdArgs string
-	pflag.VisitAll(func(f *pflag.Flag) {
-		if !f.Changed || f.Name == "verify-only" {
-			return
-		}
-		cmdArgs += fmt.Sprintf("--%s=%s ", f.Name, f.Value)
-	})
-	return fmt.Sprintf("\n// This file was automatically generated by informer-gen with arguments: %s\n\n", cmdArgs)
+	return fmt.Sprintf("\n// This file was automatically generated by informer-gen\n\n")
```
Comment (deads2k) [inline cmd/libs/go2idl/informer-gen/generators/packages.go]: @ncdc I don't care, do you?

File: pkg/apis/authorization/install/install.go [hunk 2]
```diff
@@ -43,6 +44,7 @@ func Install(groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *r
 			AddInternalObjectsToScheme: authorization.AddToScheme,
 		},
 		announced.VersionToSchemeFunc{
+			v1.SchemeGroupVersion.Version:      v1.AddToScheme,
```
Comment (deads2k) [inline pkg/apis/authorization/install/install.go]: need to fix up the version preference order above.

**Comment by deads2k** [comment]
Is there a simple way for you to confirm that the preference order is correct in discovery?

**Comment by deads2k** [comment]
minor (but important) comment.  lgtm otherwise.

File: cmd/libs/go2idl/informer-gen/generators/packages.go (hunk 1 above)
Comment (ncdc) [inline cmd/libs/go2idl/informer-gen/generators/packages.go]: No, I already gave the ok in #40700

File: pkg/apis/authorization/install/install.go (hunk 2 above)
Comment (liggitt) [inline pkg/apis/authorization/install/install.go]: fixed

**Comment by liggitt** [comment]
@k8s-bot bazel test this

**Comment by liggitt** [comment]
@mikedanese, any clue what the issue is causing this bazel error:
```
W0201 14:25:24.575] ERROR: /workspace/k8s.io/kubernetes/BUILD.bazel:35:1: Target '//pkg:all-srcs' is not visible from target '//:all-srcs'. Check the visibility declaration of the former target if you think the dependency is legitimate.
W0201 14:25:24.894] ERROR: Analysis of target '//hack:verify-boilerplate' failed; build aborted.
W0201 14:25:24.894] ____Elapsed time: 13.606s
W0201 14:25:24.895] ERROR: Couldn't start the build. Unable to run tests.
W0201 14:25:24.897] make: *** [bazel-test] Error 1
I0201 14:25:24.997] Makefile:501: recipe for target 'bazel-test' failed
E0201 14:25:25.050] Build failed
I0201 14:25:25.050] process 468 exited with code 2 after 9.0m
E0201 14:25:25.050] FAIL: pull-kubernetes-bazel
```

File: pkg/BUILD [hunk 3]
```diff
@@ -1,7 +1,3 @@
-package(default_visibility = ["//visibility:public"])
-
-licenses(["notice"])
-
```
Comment (mikedanese) [inline pkg/BUILD]: Not sure how this header got wiped out but you need the public visibility decleration

Comment (liggitt) [inline pkg/BUILD]: thanks

**Comment by liggitt** [comment]
#39471 flake
@k8s-bot unit test this
@k8s-bot cross build this

**Comment by liggitt** [comment]
@kubernetes/api-approvers for approval

**Comment by liggitt** [comment]
flake https://github.com/kubernetes/kubernetes/issues/38518
@k8s-bot unit test this

**Comment by smarterclayton** [comment]
I'm ok with this being promoted - it fits the requirements, we've had it baked for a long time, and we are testing and using it regularly.  @erictune can you weigh in whether you concur?

**Comment by erictune** [comment]
@cjcullen FYI.  subjectaccessreview going GA.  looked good to me.  now would be good time to comment if you had issues with it.

**Comment by cjcullen** [comment]
No worries about it from me...

**Comment by liggitt** [comment]
regenerated pb files

**Comment by liggitt** [comment]
@k8s-bot gce etcd3 e2e test this

**Pull request merged**